storage/filesystem.go

```diff
@@ -338,8 +338,11 @@ func (f *Filesystem) fsGroupForMetadata(meta metadata.Metadata) (*int64, error)
 		return nil, nil
 	}
 
-	fsGroupStr, ok := meta.VolumeContext[f.FSGroupVolumeAttributeKey]
-	if !ok {
+	fsGroupStr := meta.VolumeContext[f.FSGroupVolumeAttributeKey]
```

Comment:
Could you add a comment to note we respect the volumeAttribute over the pod configured value? We should probably add a warning message in the log or something if a pod specifies both and they use different values, as this is likely a misconfiguration on the users part.

Reply:
I think there might be a valid use case for providing both, the securityContext fsGroup is scoped across all containers and volumes, while the annotation is more granular scoped (just the volume).

Reply:
I updated the PR: added test cases and fixed a bug.